File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda i:i[0])
        out=[intervals[0]]
        for s,e in intervals[1:]:
            lastendinter=out[-1][1]
            if s<=lastendinter:
                out[-1][1]=max(lastendinter,e)
            else:
                out.append([s,e])
        return out
             

File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(),intervals.end());
        vector<vector<int>> ans;
        ans.push_back(intervals[0]);
        for(int i=1;i<intervals.size();i++){
            int s=intervals[i][0];
            int e=intervals[i][1];
            int prev=ans.back()[1];
            if(s<=prev){
                ans.back()[1]=max(e,prev);
            }
            else{
                ans.push_back({s,e});
            }
        }
        return ans;
    }
    
};